processor/src/decorators/mod.rs

```diff
@@ -71,26 +71,26 @@ where
         let root = [self.stack.get(5), self.stack.get(4), self.stack.get(3), self.stack.get(2)];
 
         // look up the node in the advice provider
-        let node = self.advice_provider.get_tree_node(root, depth, index)?;
+        let node = self.advice_provider.get_tree_node(root, &depth, &index)?;
 
-        // write the node into the advice tape with first element written last so that it can be
+        // write the node into the advice stack with first element written last so that it can be
         // removed first
```

Comment:
Let's update this to: "push the node onto the advice stack with the first element pushed last so that it can be popped first"